.github/workflows/test.yml

```diff
@@ -23,12 +23,24 @@ jobs:
         with:
           access_token: ${{ github.token }}
   build:
-    runs-on: [self-hosted, linux]
+    runs-on: custom-runner-AWS
     env:
       SCCACHE_CACHE_SIZE: "50G"
     needs: cancel
     steps:
     - uses: actions/checkout@v3
+    
+    - name: Installing Prerequisites
+      run: sudo apt-get update
+    - name: Installing Prerequisites 1
+      run: curl https://sh.rustup.rs -sSf -y | sh 
+    - name: Installing Prerequisites 2
+      run: sudo apt-get install -y git clang curl libssl-dev llvm libudev-dev
+    - name: Installing Prerequisites 3
+      run: git config --global submodule.recurse true
+    - name: Installing Prerequisites 4
+      run: make init
```

Comment:
they need to be baked in the runner image instead of install deps every single time